### PR TITLE
Fixing NodewView serializationn properties

### PR DESCRIPTION
**Pull request by jesusalvino**
### Purpose

Fix the bug https://jira.autodesk.com/browse/DYN-5446-serializing-dyn-file

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers
@QilongTang 

### FYIs
@RobertGlobant20 @filipeotero

### CODE REVIEW

**Comment by jesusalvino** [comment]
imagen

**Comment by QilongTang** [comment]
@jesusalvino Would you set the order for Node Model as well?

**Comment by jesusalvino** [comment]
> @jesusalvino Would you set the order for Node Model as well?

Done.

**Comment by QilongTang** [comment]
@jesusalvino Would you share the reason why for NodeModel, you did not use the sequence number as identifier but chose a different implementation?

**Comment by jesusalvino** [comment]
> @jesusalvino Would you share the reason why for NodeModel, you did not use the sequence number as identifier but chose a different implementation?

Sure, because the NodeModel has serialized properties that are not directly mapped to its C# class, like the "ConcreteType" property and the NodeModel serialized properties could have extra properties based on the different kind of nodes like "Code" property or "InputValue", that's why the order of its serialized properties is handable after its json format, not in the tag order as part of its property in its C# Class like the NodeView, the order of serialization of the NodeModel property is in the NodeModel.PropertiesSerializationOrder

File: src/DynamoCore/Graph/Workspaces/SerializationExtensions.cs [hunk 1]
```diff
@@ -95,5 +99,214 @@ internal static string SerializeIntegerSliderAs32BitType(string json)
...
+        /// Orders the properties of each Node according to the specify in the NodePropertiesOrder function
+        /// </summary>
+        /// <param name="metaContent"></param>
+        /// <returns></returns>
+        internal static string customOrderNodes(string metaContent)
+        {
+            try
+            {
+                string metaNodesStartKey = "\"Nodes\": [\r\n   ";
+                string metaNodesEndKey = "\r\n  ]";
```
Comment (mjkkirschner) [inline src/DynamoCore/Graph/Workspaces/SerializationExtensions.cs:115]: you should use something cross platform using system environment new line or similar

File: src/DynamoCore/Graph/Workspaces/SerializationExtensions.cs [hunk 2]
```diff
@@ -95,5 +99,214 @@ internal static string SerializeIntegerSliderAs32BitType(string json)
...
+                int flatNodesStartIndex = metaContent.IndexOf(metaNodesStartKey) + metaNodesStartKey.Length;
```
Comment (mjkkirschner) [inline src/DynamoCore/Graph/Workspaces/SerializationExtensions.cs:116]: I think this code has the potential to be very slow, have you profiled the serialization performance of a very large graph before and after this change?

File: src/DynamoCore/Graph/Nodes/NodeModel.cs [hunk 3]
```diff
@@ -2265,6 +2265,11 @@ public virtual string PrintExpression()
             return s;
         }
 
+        public static List<string> PropertiesSerializationOrder()
```
Comment (mjkkirschner) [inline src/DynamoCore/Graph/Nodes/NodeModel.cs:2268]: why is this public?

File: src/DynamoCore/Graph/Workspaces/SerializationExtensions.cs [hunk 4]
```diff
@@ -95,5 +99,214 @@ internal static string SerializeIntegerSliderAs32BitType(string json)
...
+            {
+                return metaContent;
+            }            
+        }
+
+        internal static List<string> splitFlatNodes(string flatNodes)
+        {
+            List<string> metaProperties = new List<string>();
+
+            var rxKeyProperty = new Regex(keyProperty);
```
Comment (mjkkirschner) [inline src/DynamoCore/Graph/Workspaces/SerializationExtensions.cs:147]: oh boy.

**Comment by mjkkirschner** [comment]
There is a lot of complexity in this new code, I did not finish reviewing it.

I am curious how much benefit there would be to simply using `[JsonProperty(Order = n)]` and avoiding the custom serialization ordering logic all together. 

I'm also concerned about the potential performance impact - we should test it with various size graphs if we're going to proceed with this.

**Comment by jesusalvino** [comment]
@QilongTang @mjkkirschner unfortunatelly I couldn't test the custom approach with a large .dyn file [enclosed here](https://jira.autodesk.com/browse/DYN-5446) , it stucks my Visual Studio when I try to debug it, so in order to keeping simple I switched to the order approach).

**Review by QilongTang — APPROVED** [review]

**Comment by QilongTang** [comment]
@jesusalvino Glad you were able to solve this using simple order definition.

**Pull request merged**